arch/rv32i/src/support.rs

```diff
@@ -18,12 +19,21 @@ pub unsafe fn wfi() {
     asm!("wfi" :::: "volatile");
 }
 
-/// TODO: implement
 pub unsafe fn atomic<F, R>(f: F) -> R
 where
     F: FnOnce() -> R,
 {
-    f()
+    let cached_mstatus = CSR.mstatus.extract();
```

Comment:
I feel that a better name would be `original_mstatus`.

Reply:
Works for me